solr/core/src/java/org/apache/solr/core/SolrXmlConfig.java

```diff
@@ -230,12 +228,8 @@ public static NodeConfig fromInputStream(
       substituteProps = new Properties();
     }
     try {
-      byte[] buf = IOUtils.toByteArray(is);
-      try (ByteArrayInputStream dup = new ByteArrayInputStream(buf)) {
-        XmlConfigFile config =
-            new XmlConfigFile(loader, null, new InputSource(dup), null, substituteProps);
-        return fromConfig(solrHome, config, fromZookeeper);
-      }
+      XmlConfigFile config = new XmlConfigFile(loader, null, null, substituteProps);
```

Comment:
how is the InputStream `is` used here? Maybe we can remove the whole method `fromInputStream()`?

` new XmlConfigFile(loader, null, null, substituteProps);` will throw NPE at some point because there is no name and InputStream is gone. So I think this method is never used anymore and is just a relic.

Did you look at Emma / coverage output?

Reply:
> how is the InputStream is used here?

Apparently, @heythem changed this method to not even take an InputStream.    But I don't see how it can work with those two nulls; how would the XmlConfigFile even know what to load?